### ISSUE TITLE
Tests or Examples - Documentation

### ISSUE THREAD

**Issue by DustinVenegas** [Microsoft/vso-node-api#21]
Are there any examples or tests available for this project? 

It's very straightforward so far, but `getPullRequestsByProject` appears to differ from the [API](https://www.visualstudio.com/en-us/integrate/api/git/pull-requests). `VsoClient.prototype.getRequestUrl` is appending a querystring parameter `searchCriteria={value}` instead of `status=active` when passing `{ status='active' }` into `getPullRequetsByProject` as the searchCriteria. 

For example: 
```
var gitClient = vsoWebApiClient.getGitApi();
gitClient.getPullRequestsByProject("MyProj", { status: 'active' }, null, null, null, handleSomething)
```

produces querystring `MyProj/_apis/git/pullRequests?searchCriteria=[object Object]` instead of `RadarBot/_apis/git/pullRequests?status=active`.

**Comment by bryanmacfarlane** [comment]
Yes, we need tests and examples.  The best example is our agent which uses it:

https://github.com/Microsoft/vso-agent 

Search for webapi: https://github.com/Microsoft/vso-agent/search?utf8=%E2%9C%93&q=webapi

We'll get some added soon.

REST API docs here: https://www.visualstudio.com/en-us/integrate/api/overview

**Comment by DustinVenegas** [comment]
To the spirit of my goal: what is the correct way query active pull requests using this node api? I tried search criteria and am using incorrectly, there's a different method, or it isn't supported in this API implementation.

**Comment by jrock2004** [comment]
Agreed, there needs to be more documentation on how this work. Would be nice to see a simple example of searching for a work item.

**Comment by stephenmichaelf** [comment]
Samples have been added [here](https://github.com/Microsoft/vsts-node-api/blob/master/samples/creation.ts), please let me know if they are insufficient.

**Comment by pradeepgudipati** [comment]
@stephenmichaelf , getPullRequestsByProject sample is still not there !

**Comment by stephenmichaelf** [comment]
Will add this specific one, thanks!
